Assets/Scripts/Enemy/EnemyHealth.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class EnemyHealth : MonoBehaviour
+{
+    public float startingHealth = 100;
+
+    private float currentHealth;
+
+    public float projectileDamage = 10f;
+
+    public Slider healthSlider;
+
+    public GameObject moneyPrefab;
+
+    private Animator animator;
+
+    private bool isDead;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = startingHealth;
+        animator = GetComponent<Animator>();
+
+        healthSlider.gameObject.SetActive(false);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (isDead) {
+            return;
+        }
+
+        if (CalculateDistanceFromPlayer() < 20)
```

Comment:
If there's a horde of zombies, there will be health bars everywhere. Let's show health bar only if the zombie took damage, so
`if (currentHealth < startingHealth && CalculateDistanceFromPlayer() < 20)`